app/controllers/website/pages_controller.rb

```diff
@@ -1,14 +1,18 @@
 #This pages controller serve static pages of the website that require no prior authentication 
 # It renders the home page, about page, contact and pricing pages...
 class Website::PagesController < ApplicationController
-  
-  def home
-  end
+    def show 
+      if valid_page? 
+       render template: "website/pages/#{params[:page]}"
+      else  
+        render file: "#{Rails.root}/public/404.html", status: :not_found
+      end
+    end
 
-  def about
-  end
-  
-  def pricing 
+    private 
 
-  end
+    def valid_page?
+      File.exist?(Pathname.new(Rails.root + "app/views/website/pages/#{params[:page]}.html.erb"))
```

Comment:
Considering:
* `Rails.root` returns a `Pathname`
* `Pathname` has the method `join` that returns another `Pathname`
* `Pathname` has the method `exist?`

You can do this, that seems simpler to me:

```ruby
Rails.root.join('app', 'views', 'website', 'pages', "#{params[:page]}.html.erb").exist?
```